src/utils/FormattingUtils.ts

```diff
@@ -27,14 +27,19 @@ export function toGWei(num: BigNumberish): BN {
 /**
  * Converts a stringified number into a BigNumber. If the string version of the num contains a `.` then it is a number which needs to be parsed to a string int.
  * @param num The number to parse.
+ * @param rounding The rounding method to use if the number has a decimal point. Defaults to "floor" or rounding down. Valid values are "floor", "round", and "ceil
  * @returns The parsed BigNumber.
  * @note This is a temporary function until we can backport support for decimal points to @across-protocol/sdk-v2.
  */
-export const toBN = (num: BigNumberish): BN => {
+export const toBN = (num: BigNumberish, rounding: "floor" | "round" | "ceil" = "floor"): BN => {
   // If the string version of the num contains a `.` then it is a number which needs to be parsed to a string int.
   if (num.toString().includes(".")) {
-    return BigNumber.from(parseInt(num.toString()));
+    // Resolve a rounding function from the rounding parameter.
+    const roundingFunction = Math[rounding];
+    // Parse the number to a string float and round it.
+    return BigNumber.from(roundingFunction(parseFloat(num.toString())));
```

Comment:
If I'm not mistaken, this can lose precision _on the part before the decimal_ if we convert to a javascript number. I would not want to use this function anywhere near sensitive fee calculations that must always be repeatable and correct unless we can be sure that the integer part of the calculation never loses precision.

> Unlike many other programming languages, JavaScript does not define different types of numbers, like integers, short, long, floating-point etc. JavaScript numbers are always stored as double precision floating point numbers, following the international IEEE 754 standard.

> Integers can only be represented without loss of precision in the range -253 + 1 to 253 - 1, inclusive (obtainable via [Number.MIN_SAFE_INTEGER](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MIN_SAFE_INTEGER) and [Number.MAX_SAFE_INTEGER](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER)), because the mantissa can only hold 53 bits (including the leading 1).

To avoid precision loss in the integer portion, we can do something like (I haven't checked this):
```ts
const [integer, decimal, ...rest] = num.toString().split(".");
assert(rest.length === 0, `invalid number ${num.toString()}`);
// Resolve a rounding function from the rounding parameter.
const roundingFunction = Math[rounding];
// Parse the decimal to a string float with '0.' in front, round it, and then add it.
// This isolates the precision loss to the part _after_ the decimal.
return BigNumber.from(integer).add(BigNumber.from(roundingFunction(parseFloat(`0.${decimal}`)));
```

Note: this does not eliminate all precision loss, but the worst that precision loss in the decimal could do is cause the output to be rounded the wrong way. Not sure if this is an issue, but it should reduce the likelihood. In general, we still need to be very careful about using this function due to the possible loss of precision and unpredictable rounding.

Reply:
Done - all precision is completed via bignumber comparisons: https://github.com/across-protocol/sdk-v2/pull/378/commits/aa34de8007b4ef122919125aebbd57b2bc4b334f